chirho/dynamical/handlers/trajectory.py

```diff
@@ -23,9 +27,20 @@ def __init__(self, times: torch.Tensor):
 
         super().__init__()
 
-    def __enter__(self) -> "LogTrajectory[T]":
+    def _pyro_simulate(self, msg) -> None:
+        initial_state = msg["args"][1]
+        start_time = msg["args"][2]
+
         self.trajectory: State[T] = State()
-        return super().__enter__()
+
+        if start_time == self.times[0]:
+            # If we're starting at the beginning of the timespan, we need to log the initial state.
+            # LogTrajectory's simulate_point will log only timepoints that are greater than the start_time of each
+            # simulate_point call, which can occur multiple times in a single simulate call when there
+            # are interruptions.
+            self.trajectory = append(
```

Comment:
I think this change is right, and is already enough to make `LogTrajectory` and `Solver` commutative. In particular, since `append` is associative and `Solver` doesn't change `initial_state` or `start_time` in its `_pyro_simulate` method, as long as no outside code needs to access `self.trajectory` before the `simulate` call finishes, it shouldn't matter if `LogTrajectory`'s `_pyro_simulate` runs before or after `Solver`'s.